promql/engine_test.go

```diff
@@ -39,8 +38,7 @@ func TestMain(m *testing.M) {
 func TestQueryConcurrency(t *testing.T) {
 	maxConcurrency := 10
 
-	dir, err := ioutil.TempDir("", "test_concurrency")
-	require.NoError(t, err)
+	dir := t.TempDir()
 	defer os.RemoveAll(dir)
```

Comment:
Apologies, I thought the `test_windows` failure was a flake (as it usually is), but there is actually a problem. `defer os.RemoveAll(dir)` should be removed as well, as `t.TempDir()` takes care of deleting the files.

Reply:
emm...     I didn't notice the the tmp dir was automatically removed.